cmd/soroban-rpc/internal/methods/util.go

```diff
@@ -0,0 +1,53 @@
+package methods
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/stellar/soroban-rpc/cmd/soroban-rpc/internal/db"
+)
+
+func getProtocolVersion(
+	ctx context.Context,
+	ledgerEntryReader db.LedgerEntryReader,
+	ledgerReader db.LedgerReader,
+) (uint32, error) {
+	var protocolVersion uint32
+	readTx, err := ledgerEntryReader.NewCachedTx(ctx)
+	if err != nil {
+		return 0, err
+	}
+	defer func() {
+		_ = readTx.Done()
+	}()
```

Comment:
`defer readTx.Done()` has the same effect, right?

Reply:
yeah actually it is same

Reply:
(I think golint will complain about ignoring error from readTx.Done() with one liner tho)

Reply:
It's better to use `nolint` than adding a defer simply to ignore the error

Reply:
And even better than ignoring the error, add it to the returned error with `errors.Join()1